pkg/logging/sanitize.go

```diff
@@ -0,0 +1,30 @@
+package logging
+
+import (
+	"io"
+	"regexp"
+)
+
+type SanitizeWriter struct {
+	w  io.Writer
+	re *regexp.Regexp
+}
+
+var dockerRateLimitCredsRe = regexp.MustCompile(
+	`(?i)you have reached your pull rate limit as\s+'[^']+':.*?(?:\.|$)`,
+)
+
+func (s *SanitizeWriter) Write(p []byte) (int, error) {
+	out := s.re.ReplaceAllString(
+		string(p),
+		"You have reached your pull rate limit (credentials hidden).",
+	)
+	return s.w.Write([]byte(out))
+}
+
+func WrapIfNeeded(w io.Writer) io.Writer {
```

Comment:
i would suggest something like "WrapInSanitizeWriter", it's much clearer what we wrap and why

```suggestion
func WrapInSanitizeWriter(w io.Writer) io.Writer {
```